tools/swift-format/swift-format.cpp

```diff
@@ -0,0 +1,344 @@
+        llvm::outs() << Text.substr(From, Index - From);
+        switch (Text[Index]) {
+        case '\n':
+          llvm::outs() << "&#10;";
+          break;
+        case '\r':
+          llvm::outs() << "&#13;";
+          break;
+        case '<':
+          llvm::outs() << "&lt;";
+          break;
+        case '&':
+          llvm::outs() << "&amp;";
+          break;
+        default:
+          llvm_unreachable("Unexpected character encountered!");
+        }
+        From = Index + 1;
+      }
+      llvm::outs() << Text.substr(From);
+    }
+
+    static void outputReplacementsXML(const Replacements &Replaces) {
+      for (const auto &R : Replaces) {
+        outs() << "<replacement "
+               << "offset='" << R.getOffset() << "' "
+               << "length='" << R.getLength() << "'>";
+        outputReplacementXML(R.getReplacementText());
+        outs() << "</replacement>\n";
+      }
+    }
+
+    // Returns true on error.
+    static bool format(StringRef FileName) {
+      ErrorOr<std::unique_ptr<MemoryBuffer>> CodeOrErr =
+        MemoryBuffer::getFileOrSTDIN(FileName);
+      if (std::error_code EC = CodeOrErr.getError()) {
+        llvm::errs() << EC.message() << "\n";
+        return true;
+      }
+      std::unique_ptr<llvm::MemoryBuffer> Code = std::move(CodeOrErr.get());
+      if (Code->getBufferSize() == 0)
+        return false; // Empty files are formatted correctly.
+
+      FormatterDocument Doc(std::move(Code));
+
+      if (!Offsets.empty() || !Lengths.empty()) {
+        if (Offsets.size() != Lengths.size()) {
+          llvm::errs() << "error: number of offsets not equal to number of lengths.\n";
+          return true;
+        }
+
+        for (unsigned i=0 ; i < Offsets.size() ; i++) {
+          unsigned FromLine = Doc.getLineAndColumn(Offsets[i]).first;
+          unsigned ToLine = Doc.getLineAndColumn(Offsets[i] + Lengths[i]).first;
+          if (ToLine == 0) {
+            llvm::errs() << "error: offset + length after end of file\n";
+            return true;
+          }
+          std::ostringstream s;
+          s << FromLine << ":" << ToLine;
+          LineRanges.push_back(s.str());
+        }
+      }
+
+      if (LineRanges.empty())
+        LineRanges.push_back("1:999999");
+
+      std::string Output = Doc.memBuffer().getBuffer();
+      Replacements Replaces;
+
+      for (unsigned Range = 0 ; Range < LineRanges.size() ; Range++) {
+        unsigned FromLine, ToLine;
+        if (parseLineRange(LineRanges[Range], FromLine, ToLine)) {
+          llvm::errs() << "error: invalid <start line>:<end line> pair\n";
+          return true;
+        }
+        if (FromLine > ToLine) {
+          llvm::errs() << "error: start line should be less than end line\n";
+          return true;
+        }
+
+        for (unsigned Line = FromLine; Line<=ToLine; Line++) {
+          size_t Offset = getOffsetOfLine(Line,Output);
+          ssize_t Length = getOffsetOfLine(Line + 1,Output) - 1 - Offset;
```

Comment:
Consider writing this as: `ssize_t Length = getOffsetOfLine(Line + 1, Output) - 1 - Offset;`